pkg/cloudprovider/provider/openstack/provider.go

```diff
@@ -148,6 +148,25 @@ func (p *provider) getConfig(s v1alpha1.ProviderSpec) (*Config, *providerconfigt
 		klog.V(6).Infof("Region from configuration or environment variable not found")
 	}
 
+	// Load timeout
+	instanceReadyCheckPeriod, err := p.configVarResolver.GetConfigVarIntValueOrEnv(rawConfig.InstanceReadyCheckPeriod, "INSTANCE_READY_CHECK_PERIOD")
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("failed to get the value of \"InstanceReadyCheckPeriod\" field, error = %v", err)
+	}
+	if instanceReadyCheckPeriod < 60 {
+		instanceReadyCheckPeriod = 120
+	}
+	c.InstanceReadyCheckPeriod = time.Duration(instanceReadyCheckPeriod) * time.Second
+
+	instanceReadyCheckTimeout, err := p.configVarResolver.GetConfigVarIntValueOrEnv(rawConfig.InstanceReadyCheckTimeout, "INSTANCE_READY_TIMEOUT")
```

Comment:
Please change this to ` p.configVarResolver.GetConfigVarIntValue` as this value can be added explicitly: 
```go
instanceReadyCheckPeriod, err := p.configVarResolver.GetConfigVarIntValue(rawConfig.InstanceReadyCheckPeriod)
```

Reply:
lgtm